libs/ledger/src/lib/ledger-export-form.tsx

```diff
@@ -0,0 +1,173 @@
+import { toNanoSeconds } from '@vegaprotocol/utils';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useEnvironment } from '@vegaprotocol/environment';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Button,
+  Loader,
+  TradingFormGroup,
+  TradingInput,
+} from '@vegaprotocol/ui-toolkit';
+import { t } from '@vegaprotocol/i18n';
+import { useAssetsDataProvider } from '@vegaprotocol/assets';
+import { localLoggerFactory } from '@vegaprotocol/logger';
+import { formatForInput } from '@vegaprotocol/utils';
+import { subDays } from 'date-fns';
+
+const DEFAULT_EXPORT_FILE_NAME = 'ledger_entries.csv';
+
+const getProtoHost = (vegaurl: string) => {
+  const loc = new URL(vegaurl);
+  return `${loc.protocol}//${loc.host}`;
+};
+
+export const LedgerExportForm = ({ partyId }: { partyId: string }) => {
+  const now = useRef(new Date());
+  const defaultFrom = subDays(now.current, 7);
+  const [dateFrom, setDateFrom] = useState(formatForInput(defaultFrom));
+  const [dateTo, setDateTo] = useState('');
+  const maxFromDate = formatForInput(new Date(dateTo || now.current));
+  const maxToDate = formatForInput(now.current);
+  const { data } = useAssetsDataProvider();
+  const assets = (data || []).reduce((aggr, item) => {
+    aggr[item.id] = item.symbol;
+    return aggr;
+  }, {} as Record<string, string>);
+  const [isDownloading, setIsDownloading] = useState(false);
+  const [assetId, setAssetId] = useState(Object.keys(assets)[0]);
+  useEffect(() => {
+    if (!assetId) {
+      setAssetId(Object.keys(assets)[0]);
```

Comment:
Its a code smell that you are setting state within a useEffect. Theres probably a better way

Reply:
don't quite understand. We need to have preselected assetId in case when assets are loaded asynchronously and we can't rely on default state of useState

Reply:
We can if we fetch assets in the ledger entries container from the app and pass in a list of assets for default state